src/components/legal/legal-modal/legal-modal.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect } from 'react';
+import { View, Platform } from 'react-native';
+import { withTheme, IThemeProps } from '../../../core/theme/with-theme';
+import stylesProvider from './styles';
+import { smartConnect } from '../../../core/utils/smart-connect';
+import { IReduxState } from '../../../redux/state';
+import { connect } from 'react-redux';
+import Modal from '../../../library/modal/modal';
+import { Legal } from '../legal';
+import { getTCVersion } from '../../../core/utils/remote-feature-config';
+import { appSetAcceptedTcVersion } from '../../../redux/app/actions';
+
+export interface IReduxProps {
+    tcVersion: number;
+    appSetAcceptedTcVersion: typeof appSetAcceptedTcVersion;
+}
+
+const mapStateToProps = (state: IReduxState) => {
+    return {
+        tcVersion: state.app.tcVersion
+    };
+};
+
+const mapDispatchToProps = {
+    appSetAcceptedTcVersion
+};
+
+export const LegalModalComponent = (
+    props: IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>
+) => {
+    const [showModal, setShowModal] = useState<boolean>(false);
+
+    useEffect(() => {
+        const tcAcceptedVersion = getTCVersion();
+
+        if (tcAcceptedVersion !== undefined) {
+            props.appSetAcceptedTcVersion(tcAcceptedVersion);
```

Comment:
i don't think this is right ... if user didn't accept t&c before, it accepts latest version automatically. T&c should be flagged as accepted only when the user taps accept button